ECIPs/ECIP-1039.md

```diff
@@ -0,0 +1,649 @@
+[ECIP1017 Monetary Policy](https://github.com/ethereumproject/ECIPs/blob/master/ECIPs/ECIP-1017.md) calculation, particularly around
+
+- block winner reward calculation
+- block winner reward for including uncles
+- block uncle miner reward for having uncles included in the winning block
+
+The content of the proposal deals exclusively with reward calculation beginning with Era 2.
+
+# Motivation
+
+
+> [@rtkacyk] In its current form, [ambiguity in the specification] may lead to different interpretations and implementations that may result in a network split in further eras. See these issues for details:
+
+- https://github.com/ethereumproject/go-ethereum/issues/352
+- https://github.com/paritytech/parity/issues/6523
+
+# Solution
+
+
+### __Block winner reward calculation.__
+
+Block winner reward calculation for a given era should be rounded down only once.
+This can be accomplished using exponentiation. In the following specifications around
+uncles, use of this strategy will be assumed as well.
+
+_eg._
+```
+eraBlockReward * 4^era / 5^era
+```
+
+__Reasoning__: a block reward in this case is _always singular_, and era number is an essentially arbitrary constant (for any given bock). If the block reward for a given era _n_ is rounded _n_ times, the iterative rounding will eventually cause the final reward to atrophy. I see no reason for magnitude of era to "degrade" the singular block reward.
+
+This, as opposed to the example below (which should NOT be used):
+
+```
+rewardReductionConstant = 4/5
+maxBlockReward = 5
+eraBlockReward = maxBlockReward # Modified below from initial max reward per era
+
+# Here, rounding would happen n times for n eras
+for (era in eras) {
+    eraBlockReward = eraBlockReward * depreciationConstant
+}
+```
+
+> See [Block winner comparison](#block-winner-reward-comparison) table for
+> expected rewards. Note that the discrepency would begin in era 22, where a single
+> rounding will yield `46116860184273879`, and iterative rounding will yield `46116860184273878`.
+
+----
+
+### Uncle rewards
+
+- Rewards for both the winning block miner for including and uncles and the miner of the included uncle(s) should round down the winning block reward prior to division by 32 in the calculation of the uncle(s) "bonus" reward.
+- The reward for uncles inclusion should reflect rounding per uncle, as opposed to rounding per "magnitude of uncles," which to say that an uncles inclusion reward is calculated using 1/32's (rounded down) of the block winner reward.
+
+#### __Block winner reward for including uncle(s).__
+
+- Uncle(s) inclusion reward should be calculated and rounded down prior to addition
+  with the block winner reward.
+
+_eg._
+```
+blockWinnerReward = blockWinnerRewardAtEra(n) # Block winner reward rounds down once prior to division
+uncleInclusionReward = floor(1/32 * blockWinnerReward) # Block uncle inclusion reward rounds down per uncle
```

Comment:
I think it's better to change this to `blockWinnerReward / 32` and then define `/` as "floor divide" because this symbol is also used above. Besides, using `1/32` might cause confusion of whether the calculation is done in floating points.

Reply:
Changed in 537451d